### PR TITLE
Turn g_mouse_binds into vector in MouseManager

### PR DESCRIPTION
**Pull request by dnnr**

### CODE REVIEW

**Comment by dnnr** [comment]
This includes #347

File: src/mouse.cpp [hunk 1]
```diff
@@ -187,14 +183,14 @@ int mouse_bind_command(int argc, char** argv, Output output) {
...
+    MouseBinding mb;
+    mb.button = button;
+    mb.modifiers = modifiers;
+    mb.action = function;
     for (int i = 3; i < argc; i++) {
-        mb->cmd.push_back(argv[i]);
+        mb.cmd.push_back(argv[i]);
     }
-    g_mouse_binds = g_list_prepend(g_mouse_binds, mb);
+    Root::get()->mouse->binds.push_back(mb);
```
Comment (dnnr) [inline src/mouse.cpp]: I was assuming here that there is no significance to the order (prepend vs. append).

Comment (t-wissmann) [reply src/mouse.cpp]: Oh right. It should be prepend actually! By preprending, I implicitely shadow an old binding to the same Modifier-Button-Combination

Comment (dnnr) [reply src/mouse.cpp]: Now using an std::list and push_front() instead

Comment (dnnr) [reply src/mouse.cpp]: I've also added a test confirming the behavior of overlapping bindings.

Comment (t-wissmann) [reply src/mouse.cpp]: Great!

**Pull request merged**